app/scripts/modules/core/src/cloudProvider/providerSelection/ProviderSelectionService.ts

```diff
@@ -47,4 +58,26 @@ export class ProviderSelectionService {
       return provider;
     });
   }
+
+  public static isDisabled(app: Application, feature: string): boolean {
+    let isDisabled = true;
+    const BreakException = {};
```

Comment:
Hmm, what are we trying to do here? I don't love the use of a custom exception for the purpose of just breaking out of a loop...I'm also a bit confused as to why we are checking if the provider is `kubernetes` on line 68.

Reply:
Refactored to use the methods **hasValue** and **getValue** instead.